DOC: make mentioning of pipeline_name consistent

yarikoptic opened the pull request:
- we really should not advocate having ```_``` as a separator, so I made example to use 'pipeline1-v1' and 'pipeline2`.  This should make it much more consistent with how we name in BIDS where '_' is a separator between entities.
- we had <pipeline> and <pipeline_name> (as we used in other places). I unified to <pipeline_name> although would not mind going to <pipeline> or introducing (expclitly) that '<pipeline>' means '<pipeline_name>[-<variant>]'

initially prompted by looking at  oddly looking example in common principles with the `pipeline_1`

yarikoptic committed:
DOC: make mentioning of pipeline_name consistent

- we really should not advocate having '_' as a separator, so I made example to use
  'pipeline1-v1' and 'pipeline2.  This should make it much more consistent
  with how we name in BIDS where '_' is a separator between entities.
- we had <pipeline> and <pipeline_name> (as we used in other places).
  I unified to <pipeline_name> although would not mind going to <pipeline>
  or introducing (expclitly) that '<pipeline>' means '<pipeline_name>[-<variant>]'

julia-pfarr reviewed (changes requested):
How about just changing `<pipeline_name>` to `<pipeline-name>` everywhere? This should make it clear that no `_` should be used anywhere in the name. Or what you suggested 

> introducing (expclitly) that '' means '<pipeline_name>[-]'

works, too. Either way, I would make it more explicit.

yarikoptic commented:
it would indeed be nice to harmonize but we ATM use both

<details>
<summary>_ : run_index,source_entities,participant_label</summary> 

```shell
❯ git grep '<[a-zA-Z0-9]*_[a-zA-Z0-9]*>' src/**/*.md | grep -v pipeline_name | nl
     1	src/CHANGES.md:-   Added `run-<run_index>` to the phase encoding maps. Improved the description.
     2	src/CHANGES.md:-   Added missing `sub-<participant_label>` in behavioral data filenames.
     3	src/appendices/hed.md:A library schema is specified in the form `<library-name<_>library-version>`.
     4	src/derivatives/common-data-types.md:                <source_entities>[_space-<space>][_desc-<label>]_<suffix>.<extension>
     5	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_res-<label>][_den-<label>][_desc-<label>]_<suffix>.<extension>
     6	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_res-<label>][_label-<label>][_desc-<label>]_mask.json
     7	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_res-<label>][_label-<label>][_desc-<label>]_mask.nii[.gz]
     8	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_seg-<label>][_res-<label>][_desc-<label>]_dseg.json
     9	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_seg-<label>][_res-<label>][_desc-<label>]_dseg.nii[.gz]
    10	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_seg-<label>][_res-<label>][_desc-<label>]_dseg.tsv
    11	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_seg-<label>][_res-<label>][_label-<label>][_desc-<label>]_probseg.json
    12	src/derivatives/imaging.md:                <source_entities>[_space-<space>][_seg-<label>][_res-<label>][_label-<label>][_desc-<label>]_probseg.nii[.gz]
    13	src/derivatives/imaging.md:                <source_entities>[_hemi-{L|R}][_space-<space>][_seg-<label>][_res-<label>][_den-<label>][_desc-<label>]_dseg.json
    14	src/derivatives/imaging.md:                <source_entities>[_hemi-{L|R}][_space-<space>][_seg-<label>][_den-<label>][_desc-<label>]_dseg.label.gii
    15	src/derivatives/imaging.md:                <source_entities>[_hemi-{L|R}][_space-<space>][_seg-<label>][_res-<label>][_den-<label>][_desc-<label>]_dseg.dlabel.nii
    16	src/derivatives/imaging.md:                <source_entities>[_hemi-{L|R}][_space-<space>][_seg-<label>][_res-<label>][_den-<label>][_desc-<label>]_dseg.tsv
    17	src/derivatives/introduction.md:    `<source_entities>[_keyword-<value>]_<suffix>.<extension>`
    18	src/pregh-changes.md:-   Added `run-<run_index>` to the phase encoding maps. Improved the description.
    19	src/pregh-changes.md:-   Added missing `sub-<participant_label>` in behavioral data filenames.

```
</details>

<details>
<summary>-: relative-path, dataset-name</summary> 

```shell
❯ git grep '<[a-zA-Z0-9]*-[a-zA-Z0-9]*>' src/**/*.md | grep -v pipeline_name | nl
     1	src/common-principles.md:bids:[<dataset-name>]:<relative-path>
     2	src/common-principles.md:which defines a `path` component of the form `<dataset-name>:<relative-path>`.
     3	src/common-principles.md:`[<dataset-name>]:<relative-path>`.
     4	src/schema/README.md:| `"bids-uri"` | A URI of the form `bids:<dataset>:<relative-path>`. If `<dataset>` is empty, the current dataset is used. | `exists('bids::participants.tsv', 'bids-uri')`                                        |

```
</details>

so which way should we go? ;-)

julia-pfarr commented:
Following Research Data Management conventions, dash should be used to represent an entity ("dataset-name") and an underscore should be used to separate entities ("dataset-name_participant-label". So I vote for using dashes

yarikoptic committed:
[DATALAD RUNCMD] Rename <pipeline_name> into <pipeline-name> for uniformity

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "git-sedi pipeline_name pipeline-name",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
[DATALAD RUNCMD] Rename <source_entities> into <source-entities> for uniformity

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "git-sedi source_entities source-entities",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^
